### PR TITLE
H7 fixes 1

### PR DESCRIPTION
**Pull request by hydra**

### CODE REVIEW

File: src/main/startup/startup_stm32h743xx.s [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+/* Zero fill the fastram_bss segment. */
+FillZerofastram_bss:
+  movs  r3, #0
+  str  r3, [r2], #4
+
+LoopFillZerofastram_bss:
+  ldr  r3, = _efastram_bss
+  cmp  r2, r3
+  bcc  FillZerofastram_bss
+/*-----*/
```
Comment (digitalentity) [inline src/main/startup/startup_stm32h743xx.s:134]: Missing stack watermaking 

```
/* Mark the heap and stack */
    ldr	r2, =_heap_stack_begin
    b	LoopMarkHeapStack

MarkHeapStack:
	movs	r3, 0xa5a5a5a5
	str	r3, [r2], #4

LoopMarkHeapStack:
	ldr	r3, = _heap_stack_end
	cmp	r2, r3
	bcc	MarkHeapStack
```

**Comment by digitalentity** [comment]
Nice, thanks @hydra. I'll merge this and do some fixes afterwards.

**Comment by hydra** [comment]
yup, there's bound to be some missing bits here and there, we'll likely find more as we do more testing.  hopefully the stuff I have done are useful steps in the right direction.

**Comment by hydra** [comment]
Current state is that SPRACINGH7EXTREME target compiles, but does not link yet, using:

```
make TARGET=SPRACINGH7EXTREME DEBUG=INFO V=1 TARGET_FLASH_SIZE=1024
```

For the SPRACINGH7EXTREME target commits, see the branch in https://github.com/spracing/inav/tree/spracingh750-initial-support which is is the same as this branch with just the SPRACINGH7EXTREME target definition added.

The code is obviously all untested, but it's progress!  Many of the topics/commits covered in this PR were also found when doing the H7 work in BF so I'm familiar with many of them.

Not quite sure how you want to handle the PR's for these commits...

**Comment by digitalentity** [comment]
I'll merge this and continue on the main branch

**Comment by hydra** [comment]
> I'll merge this and continue on the main branch

cool!  not sure it was the best idea to squash all the commits,  probably was worth cherry-picking some commits and squashing the rest.  But *great* to see more progress on this today.

Can you get someone at Matek to send me one of their new H7 boards to me?

**Pull request merged**